pandas/tests/frame/test_alter_axes.py

```diff
@@ -28,244 +25,287 @@
                         'B': ci})
         idf = df.set_index('B')
-        str(idf)
-        tm.assert_index_equal(idf.index, ci, check_names=False)
-        assert idf.index.name == 'B'
-
-        idf = df.set_index('B').reset_index().set_index('B')
-        str(idf)
-        tm.assert_index_equal(idf.index, ci, check_names=False)
-        assert idf.index.name == 'B'
+        tm.assert_index_equal(idf.index, ci)
 
-        new_df = idf.reset_index()
-        new_df.index = df.B
-        tm.assert_index_equal(new_df.index, ci, check_names=False)
-        assert idf.index.name == 'B'
+        # round-trip
+        idf = idf.reset_index().set_index('B')
+        tm.assert_index_equal(idf.index, ci)
 
     def test_set_index_cast_datetimeindex(self):
         df = DataFrame({'A': [datetime(2000, 1, 1) + timedelta(i)
                               for i in range(1000)],
                         'B': np.random.randn(1000)})
 
         idf = df.set_index('A')
-        assert isinstance(idf.index, pd.DatetimeIndex)
+        assert isinstance(idf.index, DatetimeIndex)
 
+    def test_convert_dti_to_series(self):
         # don't cast a DatetimeIndex WITH a tz, leave as object
         # GH 6032
-        i = (pd.DatetimeIndex(
-            to_datetime(['2013-1-1 13:00',
-                         '2013-1-2 14:00'], errors="raise"))
-             .tz_localize('US/Pacific'))
+        idx = DatetimeIndex(to_datetime(['2013-1-1 13:00',
+                                         '2013-1-2 14:00']),
+                            name='B').tz_localize('US/Pacific')
         df = DataFrame(np.random.randn(2, 1), columns=['A'])
 
-        expected = Series(np.array([pd.Timestamp('2013-01-01 13:00:00-0800',
-                                                 tz='US/Pacific'),
-                                    pd.Timestamp('2013-01-02 14:00:00-0800',
-                                                 tz='US/Pacific')],
-                                   dtype="object"))
+        expected = Series(np.array([Timestamp('2013-01-01 13:00:00-0800',
+                                              tz='US/Pacific'),
+                                    Timestamp('2013-01-02 14:00:00-0800',
+                                              tz='US/Pacific')],
+                                   dtype="object"), name='B')
 
         # convert index to series
-        result = Series(i)
-        assert_series_equal(result, expected)
+        result = Series(idx)
+        tm.assert_series_equal(result, expected)
 
-        # assignt to frame
-        df['B'] = i
+        # assign to frame
+        df['B'] = idx
         result = df['B']
-        assert_series_equal(result, expected, check_names=False)
-        assert result.name == 'B'
+        tm.assert_series_equal(result, expected)
 
-        # keep the timezone
-        result = i.to_series(keep_tz=True)
-        assert_series_equal(result.reset_index(drop=True), expected)
+        # convert to series while keeping the timezone
```

Comment:
why did this change?

Reply:
The whole test is about converting a DatetimeIndex to a Series, so I renamed the test as such.
I renamed `i` to `idx` for better readability.

Finally, `idx.to_series(keep_tz=True)` yields:
```
B
2013-01-01 13:00:00-08:00   2013-01-01 13:00:00-08:00
2013-01-02 14:00:00-08:00   2013-01-02 14:00:00-08:00
Name: B, dtype: datetime64[ns, US/Pacific]
```
so needs an index change to fit with `expected`. I just found that using the `index`-kwarg of `to_series` is cleaner and more understandable than using `reset_index` - and since it's the *conversion* that's being tested and not the manner of the index reset, I changed it.